utils.py

```diff
@@ -0,0 +1,164 @@
+"""A collection of utility functions (mostly file reading/writing) used
+in the PathCORE analysis scripts"""
+
+import os
+
+import pandas as pd
+
+
+def load_expression_dataset(path_to_file):
+    """
+    Parameters
+    -----------
+    path_to_file : str
+        The path to the expression dataset.
+        Currently expects a tab-delimited file with row and column names,
+        where the rows are the genes and the columns are the samples.
+
+    Returns
+    -----------
+    pandas.DataFrame
+    """
+    expression_dataset = pd.read_table(path_to_file, header=0)
+    index_on = expression_dataset.columns[0]
+    expression_dataset.set_index(index_on, inplace=True)
+    expression_dataset = expression_dataset[
+        -expression_dataset.index.str.contains('?', regex=False)]
+    expression_dataset = expression_dataset.sort_index()
+    return expression_dataset
+
+
+def load_pathway_definitions(path_to_file, shorten_pathway_names=None):
+    """
+    Parameters
+    -----------
+    path_to_file : str
+        The path to the pathway definitions file.
+        Currently expects a tab-delimited file with no header and 3 columns:
+          pathway, N (number of genes), gene1;gene2;...geneN
+    shorten_pathway_names : function(str) (default=None)
+        A function that accepts a pathway name (str) and returns the shortened
+        pathway name (str). See `shorten_pathway_names.py` in `constants`.
+
+    Returns
+    -----------
+    dict(str -> set(str)), A dictionary of pathway definitions,
+        where a pathway (key) is mapped to a set of genes (value)
+    """
+    pathway_definitions = pd.read_table(
+        path_to_file, header=None,
+        names=["pw", "size", "genes"], usecols=["pw", "genes"])
```

Comment:
the KEGG definitions file that I got from Jie originally doesn't have any column names (& I wanted to link the exact script/file from the eADAGE bitbucket) so to support the file format I had to both 1) name the columns and 2) specify which columns to use.

Looking at it now, if it causes more confusion than anything else, the pathway definitions files are so small that I could remove `usecols` and not impact anything